src/Entity/Package.php

```diff
@@ -113,8 +113,9 @@ class Package
     #[ORM\Column(type: 'string', nullable: true)]
     private ?string $language = null;
 
-    #[ORM\Column(type: 'text', nullable: true)]
-    private ?string $readme = null;
+    #[ORM\OneToOne(targetEntity: PackageReadme::class, cascade: ['persist', 'remove'], orphanRemoval: true)]
```

Comment:
cascade remove doesn't seem to work when deleting the package the readme entity is left behind, need to investigate why not.

Reply:
You need `#[JoinColumn(nullable: false, onDelete: 'CASCADE')]` for DB level cascade.

Reply:
@glaubinix no. A DB-level case will delete the row in `package` when deleting the `package_readme` (which is probably not something we want)

Reply:
@Seldaek `cascade: 'remove'` will delete the PackageReadme when deleting the Package, **using the ORM** (as this is an ORM-level cascading feature). If you delete packages in SQL, this won't have any impact and you need to delete the `package_readme` manually.

But there is a case where you need to delete only the PackageReadme: when assigning `$this->readme = null;` in `setReadme`. Maybe `orphanRemoval` works for that case but not sure.

Reply:
oh right, this only works if you have the `OneToOne` definition on the `PackageReadme`

Reply:
I think this won't help much. Won't Doctrine try to fetch the readme entity every time you load a package entity because it needs to resolve the readme field?

Reply:
> `cascade: 'remove'` will delete the PackageReadme when deleting the Package, **using the ORM** (as this is an ORM-level cascading feature).

yeah except I tried exactly that and it doesn't work.



> Maybe `orphanRemoval` works for that case but not sure

That works when setting null, which I guess I'll do as well before deleting the package with the ORM then..



> Won't Doctrine try to fetch the readme entity every time you load a package entity because it needs to resolve the readme field?

Yes that is why I did this whole mess this way, because I don't want it to eagerly fetch the readme always.